src/pudl/output/ferc1.py

```diff
@@ -1399,3 +1402,341 @@ def remove_intra_table_calculated_values(exploded: pd.DataFrame) -> pd.DataFrame
         )
     ]
     return exploded
+
+
+################################################################################
+# XBRL Calculation Tree
+################################################################################
+class XbrlCalculationTreeFerc1(BaseModel):
+    """A FERC Form 1 XBRL calculation tree.
```

Comment:
Can you say more about what information you feel like is missing / incorrect / opaque / surprising here?  The overall data structure is built around the calculations, which refer to `(table, fact)` tuples as their elements -- and those tuples are the unique identifiers for the nodes in the tree. Because we want to uniquely identify a root node and the associated leaf nodes, we have to use a directed graph and we have to identify which node we're using as the root of the tree -- otherwise you can select any node that is only connected to a single other node, and declare it the root, and still have a valid tree (structurally).

Is there another thing other than the factoids that you are imagining could potentially have been the basis of the tree? What does including factoid help disambiguate?